simulator-core/src/main/java/com/evbox/everon/ocpp/simulator/station/handlers/ocpp/GetBaseReportRequestHandler.java

```diff
@@ -0,0 +1,64 @@
+package com.evbox.everon.ocpp.simulator.station.handlers.ocpp;
+
+import com.evbox.everon.ocpp.simulator.station.StationMessageSender;
+import com.evbox.everon.ocpp.simulator.station.component.StationComponentsHolder;
+import com.evbox.everon.ocpp.v20.message.station.GetBaseReportRequest;
+import com.evbox.everon.ocpp.v20.message.station.GetBaseReportResponse;
+import com.evbox.everon.ocpp.v20.message.station.ReportDatum;
+
+import javax.annotation.Nullable;
+import java.time.Clock;
+import java.time.ZonedDateTime;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import static com.evbox.everon.ocpp.v20.message.station.GetBaseReportResponse.Status.ACCEPTED;
+import static com.evbox.everon.ocpp.v20.message.station.GetBaseReportResponse.Status.NOT_SUPPORTED;
+import static java.time.ZonedDateTime.ofInstant;
+import static java.util.Collections.singletonList;
+
+public class GetBaseReportRequestHandler implements OcppRequestHandler<GetBaseReportRequest> {
+
+    private final StationMessageSender stationMessageSender;
+    private final StationComponentsHolder stationComponentsHolder;
+    private final Clock clock;
+
+    public GetBaseReportRequestHandler(Clock clock, StationComponentsHolder stationComponentsHolder, StationMessageSender stationMessageSender) {
+        this.stationMessageSender = stationMessageSender;
+        this.stationComponentsHolder = stationComponentsHolder;
+        this.clock = clock;
+    }
+
+    @Override
+    public void handle(String callId, GetBaseReportRequest request) {
+        switch (request.getReportBase()) {
+            case FULL_INVENTORY:
+                stationMessageSender.sendCallResult(callId, new GetBaseReportResponse().withStatus(ACCEPTED));
+                sendNotifyReportRequests(request.getRequestId(), false);
+                break;
+            case CONFIGURATION_INVENTORY:
+                stationMessageSender.sendCallResult(callId, new GetBaseReportResponse().withStatus(ACCEPTED));
+                sendNotifyReportRequests(request.getRequestId(), true);
+                break;
+            default: // SUMMARY_INVENTORY
+                stationMessageSender.sendCallResult(callId, new GetBaseReportResponse().withStatus(NOT_SUPPORTED));
+        }
+    }
+
+    private void sendNotifyReportRequests(@Nullable Integer requestId, boolean onlyMutableVariables) {
+        List<ReportDatum> reportData = stationComponentsHolder.generateReportData(onlyMutableVariables);
+
+        int size = reportData.size();
+        List<Integer> sendOrder = IntStream.range(0, size).boxed().collect(Collectors.toList());
+        Collections.shuffle(sendOrder);
```

Comment:
Why do you need shuffling here? Spec says the requests have to be send async

Reply:
I agree that shuffling is rather confusing here. If it would be ever needed we can introduce it as an optional configurable action.

Reply:
sending async, is the same as sending sync in randomized order. Pretty sure we agreed to use sync sending in randomized order when we discussed this because we are a simulator, but ok i'll add async send and remove shuffling.

Reply:
it is async by it's nature. csms sends getbasereport and station replies asynchronously with notification request. no need to add any additional logic

Reply:
done